src/api/c/blas.cpp

```diff
@@ -31,16 +41,74 @@ static inline af_array dot(const af_array lhs, const af_array rhs,
     return getHandle(detail::dot<T>(getArray<T>(lhs), getArray<T>(rhs), optLhs, optRhs));
 }
 
-af_err af_matmul(   af_array *out,
-                    const af_array lhs, const af_array rhs,
-                    const af_mat_prop optLhs, const af_mat_prop optRhs)
+af_err af_sparse_matmul(af_array *out,
+                        const af_array lhs, const af_array rhs,
+                        const af_mat_prop optLhs, const af_mat_prop optRhs)
 {
     using namespace detail;
 
     try {
-        ArrayInfo lhsInfo = getInfo(lhs);
+        common::SparseArrayBase lhsInfo = getSparseArrayBase(lhs);
```

Comment:
if `lhs` is not sparse, this can create segmentation faults. You need to check for `isSparse` first before calling `getSparseArrayBase`